src/executionFiles/bridges/anyswap.ts

```diff
@@ -155,35 +157,60 @@ const getTransferStatus = async (txHash: string) => {
 //   }
 // }
 
-const parseReceipt = (tx: TransactionResponse, receipt: TransactionReceipt) => {
+const parseReceipt = (
+  toAddress: string,
+  toTokenAddress: string,
+  tx: TransactionResponse,
+  receipt: TransactionReceipt
+): ParsedReceipt => {
   const result = {
     fromAmount: '0',
     toAmount: '0',
+    toTokenAddress: toTokenAddress,
     gasUsed: '0',
     gasPrice: '0',
     gasFee: '0',
   }
-  // const decoder = new ethers.utils.AbiCoder()
 
   // gas
   result.gasUsed = receipt.gasUsed.toString()
   result.gasPrice = tx.gasPrice?.toString() || '0'
   result.gasFee = receipt.gasUsed.mul(result.gasPrice).toString()
 
-  // log
-  // const boondedLog = receipt.logs.find((log) => log.address === receipt.to) // info about initial funds
-  // const receivedLog = receipt.logs[2] // info about received funds
-  // if (boondedLog) {
-  //   const parsed = decoder.decode(bondedContractTypes, boondedLog.data) as unknown as BondedSwapped
-  //   result.fromAmount = parsed.amount.toString()
-  // }
-  // if (receivedLog) {
-  //   const parsed = decoder.decode(
-  //     receivedContractTypes,
-  //     receivedLog.data,
-  //   ) as unknown as ReceivedSwapped
-  //   result.toAmount = parsed.value.toString()
-  // }
+  // logs
+  // > Relay
+  const abiSwap = [
+    'event LogAnySwapIn(bytes32 indexed txhash, address indexed token, address indexed to, uint amount, uint fromChainID, uint toChainID)',
+  ]
+  const interfaceSwap = new ethers.utils.Interface(abiSwap)
+  receipt.logs.forEach((log) => {
+    try {
+      const parsed = interfaceSwap.parseLog(log)
+      const amount = parsed.args.amount as BigNumber
+      result.toAmount = amount.toString()
+    } catch (e) {
+      // find right log by trying to parse them
```

Comment:
Does that mean the only way to find out if that receipt log belongs to a swap or a transfer is by trying to parse it with the correct ABI and if it succeeds it is of this type?